superset/views/core.py

```diff
@@ -2704,13 +2705,19 @@ def _get_sqllab_payload(user_id: int) -> Dict[str, Any]:
                 }
                 for database in db.session.query(models.Database).all()
             }
-            # return all user queries associated with existing SQL editors
-            user_queries = (
-                db.session.query(Query)
-                .filter_by(user_id=user_id)
-                .filter(Query.sql_editor_id.cast(Integer).in_(tab_state_ids))
```

Comment:
@nytai what if you do this:

```python
tab_state_ids = [str(tab_state[0]) for tab_state in tabs_state]
...
user_queries = (
    db.session.query(Query)
    .filter_by(user_id=user_id)
    .filter(Query.sql_editor_id.in_(tab_state_ids))
```

Reply:
Haha, that definitely works! Earlier I was trying to figure out how to craft some regex to filter out rows where sql_editor_id is not numeric... casting the ids to strings makes more sense 😄